app/src/main/java/com/google/droidjump/AchievementsManager.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.droidjump;
+
+import static com.google.droidjump.R.string.achievement_great_start;
+import android.util.Log;
+import android.widget.Toast;
+import com.google.android.gms.games.achievement.Achievement;
+import com.google.android.gms.games.achievement.AchievementBuffer;
+import com.google.droidjump.models.LevelManager;
+import java.util.ArrayList;
+import java.util.Objects;
+
+/**
+ * Manages achievements unlocking.
+ */
+public class AchievementsManager {
+    MainActivity activity;
+    boolean incrementalAchievementsDataChanged;
+    ArrayList<Achievement> achievementsList;
+
+    public AchievementsManager(MainActivity activity) {
+        this.activity = activity;
+        incrementalAchievementsDataChanged = true;
+        achievementsList = new ArrayList<>();
+    }
+
+    public void checkIncrementalAchievementsChanges() {
+        if (incrementalAchievementsDataChanged) {
+            activity.getAchievementsClient().load(false).addOnCompleteListener(activity, task -> {
+                if (task.isSuccessful()) {
+                    AchievementBuffer achievementBuffer = task.getResult().get();
+                    for (Achievement achievement : achievementBuffer) {
+                        achievementsList.add(achievement.freeze());
+                    }
+                    achievementBuffer.close();
+                    incrementalAchievementsDataChanged = false;
+                    updateIncrementalAchievements();
+                } else {
+                    //TODO(dnikolskaia): Improve exception handling behavior.
+                    String message = Objects.requireNonNull(task.getException()).getMessage();
+                    Log.e("AchievementsManager", "Failed to load achievements from client: " + message);
+                    Toast.makeText(activity, "Oops, something went wrong with getting achievements", Toast.LENGTH_SHORT).show();
+                }
+            });
+        } else {
+            updateIncrementalAchievements();
```

Comment:
It is very hard to see what this "else" is referring to. Let's invert the behaviour: 
```
if (!incrementalAchievementsDataChanged) {
  updateIncrementalAchievements();
  return; 
}
// continue doing stuff
```
This will also reduce the level of nesting.

Reply:
Done